Sources/NIOHTTP2/HTTP2PipelineHelpers.swift

```diff
@@ -327,6 +327,65 @@ extension Channel {
         }
     }
 
+    /// Configures a `ChannelPipeline` to speak HTTP/2 and wraps any created inbound stream channels in `NIOAsyncChannel`s
+    /// expecting standard HTTP/2 frame types.
+    ///
+    /// In general this is not entirely useful by itself, as HTTP/2 is a negotiated protocol. This helper does not handle negotiation.
+    /// Instead, this simply adds the handler required to speak HTTP/2 after negotiation has completed, or when agreed by prior knowledge.
+    /// Use this function to setup a HTTP/2 pipeline if you wish to use `NIOAsyncChannel`s to provide async sequence abstractions
+    /// over inbound and outbound streams whilst handling back-pressure and you have no need to change the inbound/outbound types of the stream channel pipelines.
+    ///
+    /// This convenience adds the restriction that the final handler added to the stream channel by the `inboundStreamInitializer`
+    /// must have `HTTP2Frame.Payload` as its `InboundOut` and `OutboundIn` types.
+    ///
+    /// Using this rather than implementing a similar function yourself allows that pipeline to evolve without breaking your code.
+    ///
+    /// - Parameters:
+    ///     - mode: The mode this pipeline will operate in, server or client.
+    ///     - connectionConfiguration: The settings that will be used when establishing the connection. These will be sent to the peer as part of the
+    ///         handshake.
+    ///     - streamConfiguration: The settings that will be used when establishing new streams. These mainly pertain to flow control.
+    ///     - streamDelegate: The delegate to be notified in the event of stream creation and close.
+    ///     - position: The position in the pipeline into which to insert the `NIOHTTP2Handler`.
+    ///     - inboundStreamInitializer: A closure that will be called whenever the remote peer initiates a new stream.
+    /// - Returns: An `EventLoopFuture` containing the `AsyncStreamMultiplexer` inserted into this pipeline which wraps
+    /// inbound streams as `NIOAsyncChannels` after initialization. The multiplexer can be used to initiate new streams
+    /// and iterate over inbound HTTP/2 stream channels.
+    @available(macOS 10.15, iOS 13.0, watchOS 6.0, tvOS 13.0, *)
+    @_spi(AsyncChannel)
+    public func configureAsyncHTTP2Pipeline(
+        mode: NIOHTTP2Handler.ParserMode,
+        connectionConfiguration: NIOHTTP2Handler.ConnectionConfiguration,
+        streamConfiguration: NIOHTTP2Handler.StreamConfiguration,
+        streamDelegate: NIOHTTP2StreamDelegate? = nil,
+        position: ChannelPipeline.Position = .last,
+        inboundStreamInitializer: @escaping NIOHTTP2Handler.StreamInitializer
+    ) throws -> EventLoopFuture<NIOHTTP2Handler.AsyncStreamMultiplexer<NIOAsyncChannel<HTTP2Frame.FramePayload, HTTP2Frame.FramePayload>>> {
```

Comment:
I'm not sure this adds any value over this: https://github.com/apple/swift-nio-http2/blob/1ba485c5507a9c807cca0cf9432547ebca3a51ab/Sources/NIOHTTP2/HTTP2PipelineHelpers.swift#L291-L300 

It's effectively the same but assumes the stream types to be `HTTP2Frame.FramePayload` but the user can still mess up the types with the initialiser. There's maybe a small amount of value in offering this _without_ the stream initialiser but I'm actually not convinced it's worth having.

Reply:
I could be convinced otherwise but I thought that these would be useful for people who wanted to just go straight into async-land without having to deal with any of the complexity. To that end I wondered about removing the stream initializer too. Maybe such a user doesn't exist though.

Reply:
Yeah I agree with George. I would leave this out for now and see if this is requested in the future.